src/tests/token/test_dual1155_receiver.cairo

```diff
@@ -0,0 +1,149 @@
+    )
+}
+
+fn setup_non_erc1155_receiver() -> DualCaseERC1155Receiver {
+    let calldata = ArrayTrait::new();
+    let target = utils::deploy(NonImplementingMock::TEST_CLASS_HASH, calldata);
+    DualCaseERC1155Receiver { contract_address: target }
+}
+
+fn setup_erc1155_receiver_panic() -> (DualCaseERC1155Receiver, DualCaseERC1155Receiver) {
+    let snake_target = utils::deploy(
+        SnakeERC1155ReceiverPanicMock::TEST_CLASS_HASH, ArrayTrait::new()
+    );
+    let camel_target = utils::deploy(
+        CamelERC1155ReceiverPanicMock::TEST_CLASS_HASH, ArrayTrait::new()
+    );
+    (
+        DualCaseERC1155Receiver { contract_address: snake_target },
+        DualCaseERC1155Receiver { contract_address: camel_target }
+    )
+}
+
+//
+// snake_case target
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_on_erc1155_received() {
+    let (dispatcher, _) = setup_snake();
+    assert(
+        dispatcher
+            .on_erc1155_received(
+                OPERATOR(), OWNER(), TOKEN_ID, TOKEN_VALUE, DATA(true)
+            ) == IERC1155_RECEIVER_ID,
+        'Should return interface id'
+    );
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ENTRYPOINT_NOT_FOUND',))]
+fn test_dual_no_on_erc1155_received() {
+    let dispatcher = setup_non_erc1155_receiver();
+    dispatcher.on_erc1155_received(OPERATOR(), OWNER(), TOKEN_ID, TOKEN_VALUE, DATA(true));
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Some error', 'ENTRYPOINT_FAILED',))]
+fn test_dual_on_erc1155_received_exists_and_panics() {
+    let (dispatcher, _) = setup_erc1155_receiver_panic();
+    dispatcher.on_erc1155_received(OPERATOR(), OWNER(), TOKEN_ID, TOKEN_VALUE, DATA(true));
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_on_erc1155_batch_received() {
+    let (dispatcher, _) = setup_snake();
+    let token_ids = array![TOKEN_ID, TOKEN_ID];
+    let values = array![TOKEN_VALUE, TOKEN_VALUE];
+    assert(
+        dispatcher
+            .on_erc1155_batch_received(
+                OPERATOR(), OWNER(), token_ids.span(), values.span(), DATA(true)
+            ) == IERC1155_RECEIVER_ID,
+        'Should return interface id'
+    );
+}
+
+
+//
+// camelCase target
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_onERC1155Received() {
+    let (dispatcher, _) = setup_camel();
+    assert(
+        dispatcher
+            .on_erc1155_received(
+                OPERATOR(), OWNER(), TOKEN_ID, TOKEN_VALUE, DATA(true)
+            ) == IERC1155_RECEIVER_ID,
+        'Should return interface id'
+    );
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Some error', 'ENTRYPOINT_FAILED',))]
+fn test_dual_onERC1155Received_exists_and_panics() {
+    let (_, dispatcher) = setup_erc1155_receiver_panic();
+    dispatcher.on_erc1155_received(OPERATOR(), OWNER(), TOKEN_ID, TOKEN_VALUE, DATA(true));
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_onERC1155BatchReceived() {
```

Comment:
do we want to test the panic version of this one?

Reply:
Yes, as we have it for erc-721 `test_dual_onERC721Received_exists_and_panics`